### ISSUE TITLE
[Feature Request]change method bypass cloudflare 

### ISSUE THREAD

**Issue by wolfkingal2000** [AriaMoradi/Tachidesk#1174]
Please change method bypass cloudflare in FlareSolverr dont work 1 time for me if can change bypassing  to manually  like mihon can very good like newtoki site can't bypass by 
FlareSolverr and this dont updated method please and thanks

**Comment by Robonau** [comment]
there is no way to bypass cloudflare other than FlareSolverr right now

**Comment by wolfkingal2000** [comment]
> there is no way to bypass cloudflare other than FlareSolverr right now

Can't it work like on Android, where a popup or a browser opens, and we can click on it ourselves to bypass it? Normally, I can handle it myself, but I'm trying to use FlareSolverr, and it keeps saying it failed and doesn't accomplish anything.

**Comment by Robonau** [comment]
no, the only real way is to make an extension to yoink cookies from an actual browser like what haruneko is doing, but its quite allot work to do that

**Comment by wolfkingal2000** [comment]
> no, the only real way is to make an extension to yoink cookies from an actual browser like what haruneko is doing, but its quite allot work to do that


Can you tell me how to check if FlareSolverr is working or not? Also, is it possible to use my own cookies instead?

**Comment by Robonau** [comment]
1. no
2. no

**Comment by wolfkingal2000** [comment]
> 1. no
> 2. no

Alright, let's keep this topic open for now. If I find a method or if you come across one, we can see if it’s usable and share it here.

**Comment by Berobad** [comment]
Well there is https://github.com/yoori/flare-bypasser it should be compatible with flaresolverr, but at least for me it doesn't work with suwayomi

**Comment by Robonau** [comment]
> Well there is https://github.com/yoori/flare-bypasser it should be compatible with flaresolverr, but at least for me it doesn't work with suwayomi

if the api is the same (it says its compatible) and it works (no idea if it does) then the issue is you set it up incorrectly

**Comment by Shana4261** [comment]
I wish there was a solution for this issue too. I love suwayomi but since bato.to added cloudflare, I cannot use it anymore.
I tried flaresolverr and flare-bypasser but with no luck for now, still searching for a workaround

**Comment by Kickunio** [comment]
@Shana4261 try byparr - it worked for me and bypassed bato.to cloudflare protection

**Comment by Shana4261** [comment]
> @Shana4261 try byparr - it worked for me and bypassed bato.to cloudflare protection

Thanks, but unfortunately, it does not work for me, I cannot browse the extension nor download chapters

**Comment by Kickunio** [comment]
![image](https://github.com/user-attachments/assets/a73efa2b-c9eb-49a2-9ca5-8ba0068e942f)
These are some of the envs that my byparr uses (and I thought they are default). Try them, maybe they will make byparr work.

**Comment by Shana4261** [comment]
> ![image](https://private-user-images.githubusercontent.com/83494636/398224755-a73efa2b-c9eb-49a2-9ca5-8ba0068e942f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ5NzYwNzIsIm5iZiI6MTczNDk3NTc3MiwicGF0aCI6Ii84MzQ5NDYzNi8zOTgyMjQ3NTUtYTczZWZhMmItYzllYi00OWEyLTljYTUtOGJhMDA2OGU5NDJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIzVDE3NDI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlYTYxYzg2ZmEwOTMyZjk0MzE5NDA1NTMzMmYzN2Q5OTA0MWM2ZmU4YWQ4MGEwY2JlMDcxZWRlNDcxNWMxMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HadlvJfSIvw4wrswj25BYwezYf0fGAuY7kP3QUZhjy4) These are some of the envs that my byparr uses (and I thought they are default). Try them, maybe they will make byparr work.

I could make it work, the docker version was not working but the local one yes, thank you, you're a life savior

**Comment by Kickunio** [comment]
> > ![image](https://private-user-images.githubusercontent.com/83494636/398224755-a73efa2b-c9eb-49a2-9ca5-8ba0068e942f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ5NzYwNzIsIm5iZiI6MTczNDk3NTc3MiwicGF0aCI6Ii84MzQ5NDYzNi8zOTgyMjQ3NTUtYTczZWZhMmItYzllYi00OWEyLTljYTUtOGJhMDA2OGU5NDJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIzVDE3NDI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlYTYxYzg2ZmEwOTMyZjk0MzE5NDA1NTMzMmYzN2Q5OTA0MWM2ZmU4YWQ4MGEwY2JlMDcxZWRlNDcxNWMxMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HadlvJfSIvw4wrswj25BYwezYf0fGAuY7kP3QUZhjy4) These are some of the envs that my byparr uses (and I thought they are default). Try them, maybe they will make byparr work.
> 
> I could make it work, the docker version was not working but the local one yes, thank you, you're a life savior

Just for reference, before you tried the local version without my envs and it didn't work?

And about the docker version: Is your suwayomi docker-based or local?

**Comment by Shana4261** [comment]
> > > ![image](https://private-user-images.githubusercontent.com/83494636/398224755-a73efa2b-c9eb-49a2-9ca5-8ba0068e942f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ5NzYwNzIsIm5iZiI6MTczNDk3NTc3MiwicGF0aCI6Ii84MzQ5NDYzNi8zOTgyMjQ3NTUtYTczZWZhMmItYzllYi00OWEyLTljYTUtOGJhMDA2OGU5NDJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIzVDE3NDI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlYTYxYzg2ZmEwOTMyZjk0MzE5NDA1NTMzMmYzN2Q5OTA0MWM2ZmU4YWQ4MGEwY2JlMDcxZWRlNDcxNWMxMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HadlvJfSIvw4wrswj25BYwezYf0fGAuY7kP3QUZhjy4) These are some of the envs that my byparr uses (and I thought they are default). Try them, maybe they will make byparr work.
> > 
> > 
> > I could make it work, the docker version was not working but the local one yes, thank you, you're a life savior
> 
> Just for reference, before you tried the local version without my envs and it didn't work?
> 
> And about the docker version: Is your suwayomi docker-based or local?

No, I first tried the docker which did not work, I could see the requests from suwayomi coming to byparr in docker but still suwayomi was not able to browse bato extension.
Then I saw your comment but I am on Windows and do not really know how to use the envs you mentionned.
As a last resort, I cloned byparr repository locally and launch main.py. I had to browse bato extension two times from suwayomi for it to work and now suwayomi is able to browse and download chapters again.

My suwayomi is local.

**Comment by Berobad** [comment]
> @Shana4261 try byparr - it worked for me and bypassed bato.to cloudflare protection

just spun up a byparr container, and it works fine, thx

**Comment by pigajiu** [comment]
> [@Shana4261](https://github.com/Shana4261) try byparr - it worked for me and bypassed bato.to cloudflare protection

This work for me ,thx!!

**Comment by Syer10** [comment]
Going to close this since the Flaresolverr api that Suwayomi uses is supported in most of these alternatives.
